lib/core/src/server/presets.js

```diff
@@ -69,7 +69,10 @@ export const resolveAddonName = (configDir, name) => {
     // eslint-disable-next-line no-empty
   } catch (err) {}
 
-  return { name, type: 'presets' };
+  return {
+    name: resolveFrom(configDir, name),
```

Comment:
This fixes the issue with dependencies not getting required on behalf of the config, however it breaks other things (see CI output) and i'm not sure why. @gaetanmaisse do you have any suggestions?

Reply:
I'm still AFK for 2 days but this is the first PR I will take a look on Monday ;)

Reply:
🤔 , interesting, CI is failing for CRA tests only, for all the other setups it looks ok. @ndelangen I think I will need to discuss with you on this one.

Reply:
@gaetanmaisse sure, I'm back from vacation starting today, hit me up on discord?

Reply:
@gaetanmaisse let's go! https://calendly.com/ndelangen/storybook